src/allreduce_robust.h

```diff
@@ -155,18 +172,23 @@ class AllreduceRobust : public AllreduceBase {
     /*! \brief current node only helps to pass data around */
     kPassData = 2
   };
+
+  enum SeqType {
+    kOR = 0,
+    KAND = 1
+  };
   /*!
    * \brief summary of actions proposed in all nodes
    *  this data structure is used to make consensus decision
    *  about next action to take in the recovery mode
    */
   struct ActionSummary {
     // maximumly allowed sequence id
-    static const int kSpecialOp = (1 << 26);
+    static const int kSpecialOp = (1 << 20);
     // special sequence number for local state checkpoint
-    static const int kLocalCheckPoint = (1 << 26) - 2;
+    static const int kLocalCheckPoint = (1 << 20) - 2;
     // special sequnce number for local state checkpoint ack signal
-    static const int kLocalCheckAck = (1 << 26) - 1;
+    static const int kLocalCheckAck = (1 << 20) - 1;
```

Comment:
why we change it to 20?

Reply:
so we change to 20, actually I plan to use u_int32_t and keep original special consts https://github.com/chenqin/rabit/commit/b71f662b31dc5867ebb526ce552a6c8120fe1e52

>This is likely cause overflow as singed int type (1 << 26) << 5 + 2^6-1(flags) > 2^31-1

Reply:
I mean the choice of 20 is pretty random, if it is just 1 bit added in flag, we didn't choose to 25?

Reply:
fixed, use uint32_t and keep offset as it were